tests/test_modeling_common.py

```diff
@@ -115,6 +115,83 @@ def _config_zero_init(config):
 TINY_T5 = "patrickvonplaten/t5-tiny-random"
 
 
+import threading  # noqa
+
+
+class CPUMemoryTracker:
```

Comment:
should I put CPUMemoryTracker in testing_utils.py before I polish this up?

Reply:
I think my question is moot.

Unfortunately, I think I have to abandon the measuring test (leaving the functional one in place), since measuring cpu memory is super fickle - the test works well on my desktop but fails on CI. 

I tried another version with `tracemalloc` but it doesn't work well either. 

If re-run the same tests in a loop I get different numbers due to memory being cached - `gc.collect()` doesn't seem to help.

I may try some more tomorrow.

Reply:
Hmm, I think I may try `/usr/bin/time -f %M` via an external process. As it gives me an RSS max peak for the whole independent process.

Reply:
The results are very peculiar:

- for a 1GB model there is no saving
- for a 10GB model, low_mem saves 1/4 of memory, 
- for a 40GB model, low_mem saves 1/2 of memory, 

but at least that explains why my memory tracking wasn't showing the saving consistently since I was using a 0.5GB model.

So what I'm doing is:
```
/usr/bin/time -f %M python -c 'from transformers import AutoModel; AutoModel.from_pretrained("bigscience/T0", low_cpu_mem_usage=True)'
43788452
/usr/bin/time -f %M python -c 'from transformers import AutoModel; AutoModel.from_pretrained("bigscience/T0", low_cpu_mem_usage=False)'
86765944
```

Reply:
Understood. It looked like a great tool though! Too bad CPU usage can't be measured without being so finnicky.